js/radio/listen.directive.js

```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+(function(){
+  angular
+  .module("songs")
+  .directive("songListen", [
+    "$state",
+    SongListenDirectiveFunction
+  ])
+  function SongListenDirectiveFunction($state){
+    return{
+      templateUrl: "js/radio/listen.html",
+      scope: {
+        song: "="
+      },
+      link: function(scope){
+        scope.song.url = "{{SongShowViewModel.song.audio_url}}";
+        // scope.song.pauseOrPlay = function(ele){
+          // var audio = angular.element(ele.srcElement);
```

Comment:
also reference the solution - https://github.com/ga-wdi-exercises/wdi_radio_lab/blob/a9c522573141b5030eb94c4e7bc5b97985797aeb/js/songs/index.controller.js#L18